pkg/cmd/run.go

```diff
@@ -220,98 +249,368 @@ func run(c *cli.Context, comp *api.Composition) (err error) {
 		planDir = ""
 	}
 
-	req := &api.RunRequest{
-		BuildGroups: buildIdx,
-		Composition: *comp,
-		Manifest:    *manifest,
-		CreatedBy: api.CreatedBy{
-			User:   cfg.Client.User,
-			Repo:   c.String("metadata-repo"),
-			Branch: c.String("metadata-branch"),
-			Commit: c.String("metadata-commit"),
+	// Execute!
+
+	// Compute priority
+	isCollecting := c.Bool("collect")
+	isMultiple := len(runIds) > 1
+	isWaiting := c.Bool("wait") || isCollecting || isMultiple
```

Comment:
I understand because I saw the implementation but it seems a little unexpected from the user perspective.

Reply:
What about I create a follow-up:

- [ ] implement server-side runs feature
  - [ ] refactors tasks to support task groups
  - [ ] refactor the supervisor to support task groups
  - [ ] refactor the request system to support task groups
  - [ ] remove the need for waiting on the client side


?

Reply:
(resolving to merge)